### DISCUSSION TITLE
Budibase REST API (Get Table Rows Specifitt Columns Only)

### DISCUSSION THREAD

**Discussion by josh99smith** [Budibase/budibase#11574] [Help] [answered]
Is it possible to search for and return specific fields using BudiBase's "Search for Rows" API request? I'm trying to query a specific column in a table and only retrieve values from that column. The [REST API docs](https://docs.budibase.com/reference/post_tables-tableid-rows-search) don't seem to cover this.

Here's my current JSON request body:

`{
  "query": {
    "string": {
      "businessName": ""
    }
  },
  "paginate": true
}
`

**Comment by ConorWebb96** [comment, marked as answer]
Hey @josh99smith,

Have you setup your headers correctly e.g. appId, tableId, API key and ensuring its set to post instead of get request. 

I've supplied some screenshots below showing all the related settings which were used to get this to work with the rest data source;
<img width="951" alt="Screenshot 2023-08-22 at 08 21 13" src="https://github.com/Budibase/budibase/assets/126772285/aa95b79b-8590-4b3e-ae8a-7a2c61603d32">
<img width="952" alt="Screenshot 2023-08-22 at 08 23 52" src="https://github.com/Budibase/budibase/assets/126772285/f3aba671-88a8-46e1-9664-5e6e4cd0fa7d">


The query I used for this looks like this, its basically the same as yours the only change I made was for readability.

```
{
  "query": {
    "string": {
      "Name": "A different test 1"
    }
  }
}
```

In the second screenshot provided above you can see this is returning relevant results.

I hope this helps!

**Comment by AlejandroFNadal** [comment]
For anybody landing here and needing a search for ids or another numeric parameter, your body should be sth like this:

```
{
  "query": {
        "equal": {
            "billNumber":7
        },
        "string":{
            "billType":"Provision"
        }
    }
}
```
